frontend/app/src/components/home/Tweet.svelte

```diff
@@ -24,21 +25,23 @@
             intersecting &&
             !$eventListScrolling &&
             tweetWrapper !== undefined &&
-            !tweetRendered &&
+            !rendering &&
             supported
         ) {
             tweetWrapper.innerHTML = "";
-            (<any>window).twttr?.widgets
-                .createTweet(tweetId, tweetWrapper, {
-                    conversation: "none",
-                    theme: $themeStore.name,
-                })
+
+            rendering = (<any>window).twttr?.widgets.createTweet(tweetId, tweetWrapper, {
+                conversation: "none",
+                theme: $themeStore.name,
+            }) as Promise<any>;
+
+            rendering
                 .then(() => {
-                    tweetRendered = true;
```

Comment:
What prevents this from rendering multiple times?

Reply:
It will only run if `rendering` is falsy which it will only be if rendering a) has not started or b) failed

Reply:
Ahhh I didn't realise setting `rendering` to undefined only happened if there was an error.